support/sys-stat.c

```diff
@@ -51,16 +51,24 @@ Mono_Posix_FromStat (struct Mono_Posix_Stat *from, void *_to)
 	to->st_atime       = from->st_atime_;
 	to->st_mtime       = from->st_mtime_;
 	to->st_ctime       = from->st_ctime_;
-#ifdef HAVE_STRUCT_STAT_ST_ATIM
+#if HAVE_STRUCT_STAT_ST_ATIMESPEC
+	to->st_atimespec.tv_sec = from->st_atime_;
+	to->st_atimespec.tv_nsec = from->st_atime_nsec;
+	to->st_mtimespec.tv_sec = from->st_mtime_;
+	to->st_mtimespec.tv_nsec = from->st_mtime_nsec;
+	to->st_ctimespec.tv_sec = from->st_ctime_;
+	to->st_ctimespec.tv_nsec = from->st_ctime_nsec;
+#else
+#    ifdef HAVE_STRUCT_STAT_ST_ATIM
 	to->st_atim.tv_nsec = from->st_atime_nsec;
-#endif
-#ifdef HAVE_STRUCT_STAT_ST_MTIM
+#    endif
```

Comment:
We do not have spaces in between `#` and `if/else/endif`

Reply:
We should, someone might have undone code that historically did that that, but it is a bad practice.